node-persistent-cache.cpp

```diff
@@ -532,10 +531,10 @@ int node_persistent_cache::get_list(nodelist_t &out, const idlist_t nds)
 {
     out.assign(nds.size(), osmNode());
 
-    bool need_fetch = false;;
+    bool need_fetch = false;
     for (size_t i = 0; i < nds.size(); ++i) {
         /* Check cache first */
-        if (ram_cache && (ram_cache->get(&out[i], nds[i]) != 0)) {
+        if (!ram_cache || (ram_cache->get(&out[i], nds[i]) == 0)) {
```

Comment:
So ti turns out, I got the semantics of ram_cache->get() wrong. It actually returns 0 when it finds a node. (Fix will follow in a bit.) The consequence is that all nodes are looked up in the flat node store. If that makes the memory hogging worse, the flat node store must be the culprit. Investigating.